cli/src/commands/branch.rs

```diff
@@ -100,9 +100,13 @@ pub struct BranchDeleteArgs {
 pub struct BranchListArgs {
     /// Show all tracking and non-tracking remote branches including the ones
     /// whose targets are synchronized with the local branches.
-    #[arg(long, short, conflicts_with_all = ["names", "revisions"])]
+    #[arg(long, short, conflicts_with_all = ["names", "revisions", "tracked"])]
     all: bool,
 
+    /// Show remote tracked branches for specified remotes
+    #[arg(long, short, conflicts_with_all = ["names", "revisions", "all"], value_name = "REMOTE")]
+    tracked: Vec<String>,
```

Comment:
I'm not sure how much more tedious it is. I wasn't particularly happy with `-t remote1 -t remote2` plus this will simplify the code much more!

That said, what do you think about omitting the git remote from `jj branch list --tracked`? Since it's always tracking.. This was the main motivation behind having it require a remote name since it seemed like the most used case would be for the origin remote.

Reply:
Oh, I misread.. I get it, we may want --remote later or just through name. In that case I'll only do tracked as a boolean and filtering different remotes can be done in another commit or just by the standard filtering.

Reply:
> That said, what do you think about omitting the git remote from `jj branch list --tracked`? Since it's always tracking..

Suppose the motivation of using `--tracked` is to list tracked (real) remote branches, it's probably okay to filter out the `@git` remote. We already have a similar special case in `remote_branches()` revset.

If we add some way to filter remotes by name, maybe we can say that `--tracked` implies `--remote=!"git"` by default? (and the user-specified remote filter will override the default.)

> we may want --remote later or just through name.

I don't have any concrete plan, but I just thought it might make sense to add `--remote` or `glob:*@<remote>`.

Reply:
Sounds good, I went ahead and added the default filtering for != git and it can be easily extended to adapt for --remote later.